questionnaire/routes.test.js

```diff
@@ -434,4 +435,110 @@ describe('V2 openapi validation', () => {
+            expect(response.body.errors[0].detail).toEqual('jwt malformed');
+        });
+
+        it('should return status code 403 if the bearer token has insufficient scope', async () => {
+            // THIS IS A TOKEN WITH A DUMMY SCOPE
+            const dummyToken =
+                'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJkYXRhLWNhcHR1cmUtc2VydmljZSIsImlzcyI6IiQuYXVkIiwianRpIjoiNTFhODljYWUtM2Q1MC00ZDc1LTliMmEtMjU2NzliODgwMTkxIiwic3ViIjoiY2ljYS13ZWIiLCJzY29wZSI6ImNyZWF0ZTpub3RoaW5nIiwiaWF0IjoxNjgwNzk4NDU5fQ.97LgtlW_dcAV0Xno6BsbVmuyhLtq4gCoVWGQ56_VmEk';
+            const response = await request(app)
+                .get('/api/questionnaires/285cb104-0c15-4a9c-9840-cb1007f098fb/progress-entries')
+                .set('Authorization', `Bearer ${dummyToken}`)
+                .set('Content-Type', 'application/vnd.api+json')
+                .set('Dcs-Api-Version', '2023-05-17')
+                .set('on-behalf-of', `urn:uuid:f81d4fae-7dec-11d0-a765-00a0c91e6bf6`);
+            expect(response.body).toHaveProperty('errors');
+            expect(response.body.errors[0].status).toEqual(403);
+            expect(response.body.errors[0].detail).toEqual('Insufficient scope');
+        });
+
+        it('should return status code 404 if the query string contains incorrect data', async () => {
+            const response = await request(app)
+                .get(
+                    '/api/questionnaires/285cb104-0c15-4a9c-9840-cb1007f098fb/progress-entries?filter[sectionId]=p--not-a-valid-section'
+                )
+                .set('Authorization', `Bearer ${token}`)
+                .set('Content-Type', 'application/vnd.api+json')
+                .set('Dcs-Api-Version', '2023-05-17')
+                .set('on-behalf-of', `urn:uuid:f81d4fae-7dec-11d0-a765-00a0c91e6bf6`);
+            expect(response.body).toHaveProperty('errors');
+            expect(response.body.errors[0].status).toEqual(404);
+            expect(response.body.errors[0].detail).toEqual(
+                'ProgressEntry "p--not-a-valid-section" does not exist'
+            );
+        });
+
+        describe('Requests made MUST include owner data', () => {
+            it('should return status code 400 if owner data is NOT included in the request header', async () => {
+                const response = await request(app)
+                    .get(
+                        '/api/questionnaires/285cb104-0c15-4a9c-9840-cb1007f098fb/progress-entries'
+                    )
+                    .set('Authorization', `Bearer ${token}`)
+                    .set('Content-Type', 'application/vnd.api+json')
+                    .set('Dcs-Api-Version', '2023-05-17');
+                expect(response.body).toHaveProperty('errors');
+                expect(response.body.errors[0].status).toEqual(400);
+                expect(response.body.errors[0].detail).toEqual(
+                    "should have required property 'on-behalf-of'"
```

Comment:
title case the header name here

Reply:
The error transforms the header to lower-case from some reason so this is correct.